### PR TITLE
Reordering and commenting in preparation for pile-ups

### PR DESCRIPTION
**Pull request by eggrobin**
Moved code around, added callbacks with commented stubs.

### CODE REVIEW

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 1]
```diff
@@ -696,23 +700,11 @@ where is_in_physics_bubble(vessel)
...
-        return;
-      } else if (plugin_time == universal_time) {
-        time_is_advancing_ = false;
-        return;
-      }
-      time_is_advancing_ = true;
       plugin_.SetMainBody(
           FlightGlobals.currentMainBody.GetValueOrDefault(
               FlightGlobals.GetHomeBody()).flightGlobalsIndex);
+
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Why two blank lines?

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 2]
```diff
@@ -828,14 +793,101 @@ where is_in_physics_bubble(vessel)
...
+  AdvanceTimeAndNudgeVesselsAfterPhysicsSimulation(double universal_time) {
+     yield return new UnityEngine.WaitForFixedUpdate();
+     // Unity's physics has just finished doing its thing.  If we correct the
+     // positions here, nobody will know that they're not the ones obtained by
+     // Unity.  Careful however: while the positions here are those of the next
+     // step, the Planetarium hasn't run yet, and still has its old time.
+     universal_time += Planetarium.TimeScale * Planetarium.fetch.fixedDeltaTime;
+
+     // The collisions are reported and stored into |currentCollisions| in
+     // OnCollisionEnter|Stay|Exit, which occured while we yielded.
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: occurred

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 3]
```diff
@@ -828,14 +793,101 @@ where is_in_physics_bubble(vessel)
...
+     // Here, the |currentCollisions| are the collisions that occured in the
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Same.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 4]
```diff
@@ -828,14 +793,101 @@ where is_in_physics_bubble(vessel)
...
+                                       vessel2.id.ToString());
+             }
+           }
+         }
+       }
+     }
+
+     double plugin_time = plugin_.CurrentTime();
+     if (plugin_time > universal_time) {
+       // TODO(Egg): Make this resistant to bad floating points up to 2ULPs,
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: egg

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

**Pull request merged**